spark/v4.0/spark-extensions/src/main/scala/org/apache/spark/sql/catalyst/parser/extensions/IcebergSqlExtensionsAstBuilder.scala

```diff
@@ -30,7 +30,7 @@ import org.apache.iceberg.NullOrder
 import org.apache.iceberg.SortDirection
 import org.apache.iceberg.expressions.Term
 import org.apache.iceberg.spark.Spark3Util
-import org.apache.spark.sql.AnalysisException
+import org.apache.spark.sql.catalyst.analysis.IcebergAnalysisException
```

Comment:
Why are we switching to our internal Exception class here?

Reply:
Because Spark 4.0 no longer allows the construction of a new `AnalysisException` with just a string message. I actually have a separate [PR](https://github.com/apache/iceberg/pull/10766) for this. We can probably merge this change in Spark 3.5, although it's not strictly required there.